drivers/scheduler/k8s/k8s.go

```diff
@@ -3019,6 +2993,63 @@ func (k *K8s) addLabelsToPVC(pvc *v1.PersistentVolumeClaim, labels map[string]st
 	}
 }
 
+// CreateAutopilotRule creates the AutopilotRule object
+func (k *K8s) CreateAutopilotRule(apRule apapi.AutopilotRule) (*apapi.AutopilotRule, error) {
+	t := func() (interface{}, bool, error) {
+		apRule.Labels = defaultTorpedoLabel
+		aRule, err := k8sAutopilot.CreateAutopilotRule(&apRule)
+		if errors.IsAlreadyExists(err) {
+			if rule, err := k8sAutopilot.GetAutopilotRule(apRule.Name); err == nil {
+				logrus.Infof("Using existing AutopilotRule: %v", rule.Name)
+				return aRule, false, nil
+			}
+		}
+		if err != nil {
+			return nil, true, fmt.Errorf("Failed to create autopilot rule: %v. Err: %v", apRule.Name, err)
+		}
+
+		return aRule, false, nil
+	}
+
+	apRuleObj, err := task.DoRetryWithTimeout(t, k8sObjectCreateTimeout, DefaultRetryInterval)
+	if err != nil {
+		return nil, err
+	}
+	logrus.Infof("Created autopilot rule: %+v", apRuleObj)
+
+	return apRuleObj.(*apapi.AutopilotRule), nil
+}
+
+// GetAutopilotRule gets the AutopilotRule for the provided name
+func (k *K8s) GetAutopilotRule(name string) (*apapi.AutopilotRule, error) {
+	return k8sAutopilot.GetAutopilotRule(name)
+}
+
+// UpdateAutopilotRule updates the AutopilotRule
+func (k *K8s) UpdateAutopilotRule(apRule apapi.AutopilotRule) (*apapi.AutopilotRule, error) {
+	aRule, err := k.GetAutopilotRule(apRule.Name)
+	if err != nil {
+		return nil, err
+	}
+	aRule.Spec = apRule.Spec
```

Comment:
the same value has been assigned, remove this line

Reply:
It is not the same value. aRule is runtime autopilot rule, and apRule is the rule which will be updated

Reply:
consider using more readable names

Reply:
could you please rename `aRule -> rule`